### PR TITLE
Use Jest for test

### PR DESCRIPTION
**Pull request by lask**

### CODE REVIEW

File: vscode/package.json [hunk 1]
```diff
@@ -288,6 +296,7 @@
     "dependencies": {
         "ovsx": "^0.2.0",
         "semver": "^7.3.5",
+        "update": "^0.4.2",
```
Comment (jstroem) [inline vscode/package.json:299]: I Think this is by accident?

File: vscode/package.json [hunk 2]
```diff
@@ -260,26 +261,33 @@
...
         "lint": "eslint src --ext ts",
         "lint-fix": "eslint src --ext ts --fix",
         "package": "vsce package --yarn",
         "clean": "rm -rf ./out ./dist ./node_modules"
     },
     "devDependencies": {
+        "@babel/core": "^7.14.6",
+        "@babel/plugin-transform-modules-commonjs": "^7.14.5",
+        "@babel/preset-env": "^7.14.7",
+        "@babel/preset-typescript": "^7.14.5",
```
Comment (jstroem) [inline vscode/package.json:274]: I don't think we use babel anywhere, try and remove the `babel.config.js` config file and these imports.

File: vscode/package.json [hunk 3]
```diff
@@ -260,26 +261,33 @@
...
         "@types/glob": "^7.1.1",
-        "@types/mocha": "^5.2.6",
+        "@types/jest": "^26.0.23",
         "@types/node": "^14.17.1",
         "@types/semver": "^7.3.5",
         "@types/vscode": "^1.55.0",
         "@typescript-eslint/eslint-plugin": "^4.24.0",
         "@typescript-eslint/parser": "^4.24.0",
+        "babel-jest": "^27.0.6",
```
Comment (jstroem) [inline vscode/package.json:282]: same, I don't think we use babel.

**Review by jstroem — APPROVED** [review]

**Review by jstroem — COMMENTED** [review]
remember to add the package-lock.json file again

**Comment by lask** [comment]
There is a `yarn.lock` having that checked out seems to ruin the install step of the Jenkins pipeline.

**Pull request merged**